ghcide/src/Development/IDE/Spans/AtPoint.hs

```diff
@@ -213,21 +214,33 @@ atPoint
   -> DocAndKindMap
   -> HscEnv
   -> Position
-  -> Maybe (Maybe Range, [T.Text])
-atPoint IdeOptions{} (HAR _ hf _ _ kind) (DKMap dm km) env pos = listToMaybe $ pointCommand hf pos hoverInfo
+  -> IO (Maybe (Maybe Range, [T.Text]))
+atPoint IdeOptions{} (HAR _ hf _ _ (kind :: HieKind hietype)) (DKMap dm km) env pos =
```

Comment:
```suggestion
atPoint IdeOptions{} (HAR _ hf _ _ kind) (DKMap dm km) env pos =
```

Or is the type annotation important?

Reply:
Yes, because `HAR` has an existential type, following signatures wont work without this, .

https://github.com/haskell/haskell-language-server/blob/47cf162ec7c7ff6a0960ccddee85f3bc31d2b268/ghcide/src/Development/IDE/Core/RuleTypes.hs#L209-L221